pkg/gardenlet/controller/shoot/shoot_control.go

```diff
@@ -187,15 +187,15 @@ func (c *Controller) reconcileShootRequest(ctx context.Context, req reconcile.Re
 		return reconcile.Result{}, err
 	}
 
-	// fetch related objects required for shoot operation
-	project, err := gutil.ProjectForNamespaceFromLister(c.k8sGardenCoreInformers.Core().V1beta1().Projects().Lister(), shoot.Namespace)
+	gardenClient, err := c.clientMap.GetClient(ctx, keys.ForGarden())
 	if err != nil {
-		return reconcile.Result{}, err
+		return reconcile.Result{}, fmt.Errorf("failed to get garden client: %w", err)
 	}
 
-	gardenClient, err := c.clientMap.GetClient(ctx, keys.ForGarden())
+	// fetch related objects required for shoot operation
+	project, _, err := gutil.ProjectAndNamespaceFromReader(ctx, gardenClient.APIReader(), shoot.Namespace)
```

Comment:
Now we always us a "direct" call to retrieve the project whereas we used a cached version via a Project lister before. Shouldn't we still use the lister and switch to the client once the cache is been graduated to beta?